code/go/0chain.net/smartcontract/zcnsc/mint.go

```diff
@@ -133,36 +135,34 @@ func (zcn *ZCNSmartContract) Mint(trans *transaction.Transaction, inputData []by
 
 	var (
 		amount currency.Coin
-		n      currency.Coin
 		share  currency.Coin
 	)
 	share, _, err = currency.DistributeCoin(gn.ZCNSConfig.MaxFee, int64(len(payload.Signatures)))
 	if err != nil {
 		err = errors.Wrap(err, fmt.Sprintf("%s, DistributeCoin operation, %s", code, info))
 		return
 	}
-	n, err = currency.Int64ToCoin(int64(len(payload.Signatures)))
+
+	amount, err = currency.MinusCoin(payload.Amount, share)
 	if err != nil {
-		err = errors.Wrap(err, fmt.Sprintf("%s, convert len signatures to coin, %s", code, info))
+		err = errors.Wrap(err, fmt.Sprintf("%s, payload.Amount - share * len(signatures), %s", code, info))
 		return
 	}
-	amount, err = currency.MinusCoin(payload.Amount, share*n)
+	payload.Amount = amount
+
+	rand.Seed(int64(trans.CreationDate.Duration()))
+	sig := payload.Signatures[rand.Intn(len(payload.Signatures))]
+
+	sp, err := zcn.getStakePool(sig.ID, ctx)
 	if err != nil {
-		err = errors.Wrap(err, fmt.Sprintf("%s, payload.Amount - share * len(signatures), %s", code, info))
+		err = errors.Wrap(err, fmt.Sprintf("failed to retrieve stake pool for authorizer %s", sig.ID))
 		return
 	}
 
-	payload.Amount = amount
-	for _, sig := range payload.Signatures {
-		err = ctx.AddMint(&state.Mint{
-			Minter:     gn.ID,
-			ToClientID: sig.ID,
-			Amount:     share,
-		})
-		if err != nil {
-			err = errors.Wrap(err, fmt.Sprintf("%s, AddMint for authorizers, %s", code, info))
-			return
-		}
+	err = sp.DistributeRewards(share, sig.ID, spenum.Authorizer, spenum.FeeRewardAuthorizer, ctx)
```

Comment:
Do we only reward one authorizer? when will other authorizers get rewarded as they all have put effort in validating the mint/burn txns.

Reply:
@dabasov suggested for now select one authorizer randomly

Reply:
yeah, we discussed it with @guruhubb , on long run it will be uniformly distributed, otherwise we need to reward every authorizer which can hit performance.